src/Microsoft.AspNetCore.Authentication/Events/HandleRequestContext.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Authentication
+{
+    public class HandleRequestContext<TOptions> : BaseContext<TOptions> where TOptions : AuthenticationSchemeOptions
+    {
+        protected HandleRequestContext(
+            HttpContext context,
+            AuthenticationScheme scheme,
+            TOptions options)
+            : base(context, scheme, options) { }
+
+        /// <summary>
+        /// The <see cref="HandleRequestResult"/> which is used by the handler.
+        /// </summary>
+        public HandleRequestResult Result { get; protected set; }
+
+        /// <summary>
+        /// Discontinue all processing for this request and return to the client.
+        /// The caller is responsible for generating the full response.
+        /// </summary>
+        public void HandleResponse() => Result = HandleRequestResult.Handle();
+
+        /// <summary>
+        /// Discontinue processing the request in the current handler.
+        /// </summary>
+        public void SkipRequest() => Result = HandleRequestResult.SkipRequest();
```

Comment:
> This is intentional as its not longer correct, returning skip does not skip to the next middleware anymore.

Ultimately, it does: https://github.com/aspnet/Security/blob/dev/src/Microsoft.AspNetCore.Authentication/AuthenticationMiddleware.cs#L44

There's fundamentally no difference between how `HandleRequestAsync` works in 1.x and how it works in 2.0: returning `true` from `HandleRequestAsync` immediately stops the request processing and returning `false` allows the authentication middleware to invoke the other handlers until one of them returns `true`: if none returns `true`, the next middleware in the pipeline is invoked.